templates/aws-cf-asg-marketplace.yaml

```diff
@@ -0,0 +1,580 @@
+      AccessProxyToken:
+        default: Access Proxy Token
+      AccessProxyPublicPort:
+        default: Access Proxy public port
+      VpcId:
+        default: Vpc Id
+      EC2Subnets:
+        default: Subnets for instances
+      NLBPublicSubnets:
+        default: Subnets for Network Load Balancer
+      EC2AssociatePublicIpAddress:
+        default: Associate Public Ip Address to instances
+      EC2ASGDesiredCapacity:
+        default: Number of desired instances
+      EC2InstanceType:
+        default: Instance type to use for the proxy
+      EC2KeyName:
+        default: SSH Public Key Name for the instance
+      AccessProxyLoglevel:
+        default: Log level for the application
+      CloudWatchLogs:
+        default: Send logs to CloudWatch
+      CloudWatchLogsRetentionInDays:
+        default: Log Retention
+
+Parameters:
+
+  AccessProxyToken:
+    Description: >-
+      CloudGen Access Proxy Token for this proxy (obtained from the console after
+      proxy creation).
+    Type: String
+    MinLength: '20'
+    NoEcho: true
+    AllowedPattern: ^https:\/\/.*\.fyde\.com\/proxies.*proxy_auth_token.*$
+
+  AccessProxyPublicPort:
+    Description: >-
+      Public port for this proxy (must match the value configured in the console for
+      the proxy).
+    Default: 443
+    Type: Number
+    MinValue: 1
+    MaxValue: 65535
+
+  VpcId:
+    Description: Select the Virtual Private Cloud (VPC) to use.
+    Type: AWS::EC2::VPC::Id
+    ConstraintDescription: >-
+      must be the VPC Id of an existing Virtual Private Cloud. Outbound traffic
+      for the default security group associated with this VPC should be enabled.
+
+  NLBPublicSubnets:
+    Description: >-
+      Select the Public Subnet Ids to use for the Network Load Balancer.
+      NOTE: Use Public Subnets only.
+    Type: List<AWS::EC2::Subnet::Id>
+
+  EC2Subnets:
+    Description: >-
+      Select the Subnet Ids to use for the EC2 instances.
+      NOTE: Use Private Subnets with NAT Gateway configured or Public Subnets.
+    Type: List<AWS::EC2::Subnet::Id>
+
+  EC2AssociatePublicIpAddress:
+    Description: >-
+      Default is false. If true, each instance receives a unique public IP address.
+      Required when the instances are deployed in a public network.
+    Type: String
+    Default: false
+    AllowedValues:
+      - false
+      - true
+
+  EC2ASGDesiredCapacity:
+    Description: Enter the desired capacity for the ASG
+    Default: 2
+    Type: Number
+    MinValue: 1
+
+  EC2KeyName:
+    Description: >-
+      EC2 instance key name
+      Optional: Selecting existing EC2 KeyPair
+    Type: AWS::EC2::KeyPair::KeyName
+
+  EC2InstanceType:
+    Description: EC2 instance type
+    Type: String
+    Default: t3.small
+    AllowedValues:
+      - t3.small
+      - t3.medium
+      - t3.large
+      - t2.small
+      - t2.medium
+      - t2.large
+
+  CloudWatchLogs:
+    Description: >-
+      Send '/var/log/message' logs to CloudWatch.
```

Comment:
@GMartinez-Sisti can we add the journalctl output for envoy and fydeproxy to cloudwatch? otherwise people still needs to SSH to get those and TBH there isn't much interesting about the instance state in /var/log/messages anyway